### PR TITLE
completed-fetch-in-react-Bishan-Webeshete

### PR DESCRIPTION
**Pull request by BishanWebeshete**
Screen Shot 2023-04-04 at 11 14 54 AM

### CODE REVIEW

File: fetch-in-react/my-app/src/User.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        try {
+          const response = await fetch(`https://jsonplaceholder.typicode.com/users/${userId}`);
+          if (!response.ok) {
+            throw new Error('Network response was not OK', response.status)
+          }
+          const jsonData = await response.json();
+          setUser(jsonData);
+          setIsLoading(false);
+        } catch (error) {
+          setError(error);
```
Comment (RobertGardner) [inline fetch-in-react/my-app/src/User.js:23]: set isLoading to false here, too, or you won't get the error message to display.

File: fetch-in-react/my-app/src/Users.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          throw new Error ('Network response was not OK', response.status)
+        }
+        const jsonData = await response.json();
+        setUsers(jsonData);
+      } catch(error) {
+        setError(error);
+      }
+     }
+     logJSONData();
+     setIsLoading(false);
```
Comment (RobertGardner) [inline fetch-in-react/my-app/src/Users.js:29]: You can't set it here or your "Loading..." message will go away immediately, while the request is still in progress.

**Review by RobertGardner — CHANGES_REQUESTED** [review]

**Comment by BishanWebeshete** [comment]
updated assignment to properly use setIsLoading.

**Review by RobertGardner — APPROVED** [review]
Your code and understanding have improved markedly in just the last couple of days. It seems like something has finally "clicked" for you!

**Pull request merged**